src/components/forms/JurisdictionSelect/index.tsx

```diff
@@ -269,6 +271,14 @@ export const handleChangeWithOptions = (
           if (fiStatusFieldName) {
             const val = optionVal.fiStatus || NOT_AVAILABLE;
             form.setFieldValue(fiStatusFieldName, val);
+            form.setFieldTouched(fiStatusFieldName, true);
+            const updatedForm = { ...form };
+            updatedForm.values.jurisdictions[0] = { id: optionVal.value, name: optionVal.label };
```

Comment:
Could you explain this line?  It seems fragile to me in that it specifically targets element 0.  What if there are more, or less?

Reply:
Confirmed with @ciremusyoka, There can be multiple jurisdictions, Updating in next commit.